src/MergeContext.js

```diff
@@ -1284,7 +1303,16 @@ class PullRequest {
             // TODO: Process Config.failedOtherLabel() PRs last and ignore their failures.
             if (!this._labels)
                 this._labels = new Labels([], this._prNumber());
-            this._labels.add(Config.failedOtherLabel());
+
+            // failedOtherLabel() and failedStagingChecksLabel() are mutually exclusive
+            if (wasStaged) {
+                this._labels.remove(Config.failedOtherLabel());
+                // avoid livelocking
+                this._labels.add(Config.failedStagingOtherLabel());
+            } else {
+                // removed by humans: Config.failedStagingOtherLabel()
+                this._labels.add(Config.failedOtherLabel());
```

Comment:
The idea was that the presence of failedStagingOtherLabel() prevents from proceeding up to here because _checkForHumanLabels() will throw earlier at the beginning of _doProcess() and the process() catch block will return after 'knownProblem' flag is checked.
However, there is a probability that _doProcess() throws before _checkForHumanLabels(). In this case, failedStagingOtherLabel may exist but there is no 'knownProblem' flag. So we can either do as you suggest, (i.e, leaving failedStagingOtherLabel as is) or remove failedStagingOtherLabel, updating the PR with a fresher failedOtherLabel (i.e., reflecting more precisely the _current_ state of the PR).

Reply:
Thank you for explaining this complicated issue.

> or remove failedStagingOtherLabel, updating the PR with a fresher failedOtherLabel

I understand the desire to reflect the current state, but our documentation implies that `M-failed-staging-other` is only cleared by humans. I do not think we should make an exception here, especially since `M-failed-other` can be automatically removed later (with an end/cumulative result being no failed labels at all!). I think the best handling of an early exception (i.e. an exception thrown before labels are obtained or checked) is to (effectively) set both labels:

```Javascript
if (wasStaged) {
    this._labels.remove(Config.failedOtherLabel()); // if any
    // avoid livelocks
    this._labels.add(Config.failedStagingOtherLabel());
} else {
    // Since knownProblem is false, either there was no failedStagingOtherLabel()
    // or the problem happened before we could check for it. In the latter case,
    // that label will remain set, and we will add failedOtherLabel(), reflecting the
    // compound nature of the problem.
    this._labels.add(Config.failedOtherLabel());
}
```

However, the above (unfortmatted) sketch exposes another concern: Why are we deleting failedOtherLabel() only in some special cases? Should not that temporary label be removed ASAP (literally), in the beginning of the PR processing? In other words, if we could not remove failedOtherLabel() back then, then we should not be able to remove it here (and should not be trying). Perhaps we should call _removeTemporaryLabelsSetByAnubis() a lot earlier, right after _loadLabels() and remove the sketched "if any" line from this code?

Please adjust the code accordingly, addressing the old concern and, if possible, the new one as well.

Reply:
> Should not that temporary label be removed ASAP (literally), in the beginning of the PR processing? 

Agreed.

> Perhaps we should call _removeTemporaryLabelsSetByAnubis() a lot earlier

No, we cannot: for example, we should leave the 'invalid message' label as is until we are able to check it again. We should remove only failedOtherLabel() there.

I adjusted according to your sketch.